tests-ng/test_duplicate_uids.py

```diff
@@ -0,0 +1,38 @@
+from typing import Dict, List
+
+from plugins.password_shadow import passwd_entries
+
+
+def test_duplicate_uids(passwd_entries: List[Dict[str, str | List[str]]]):
+    """
+    Check if any duplicate userids are defined in /etc/passwd.
+    """
+    duplicates = _check_for_duplicates(passwd_entries=passwd_entries)
+    assert (
+        len(duplicates) == 0
+    ), f"uids {duplicates} are used multiple times in /etc/passwd"
+
+
+def test_duplicate_uids_detection(passwd_entries: List[Dict[str, str | List[str]]]):
+    """
+    Appending an existing entry to the list to check if the duplication check finds this.
+    """
+    passwd_entries.append(passwd_entries[0])
+
+    duplicates = _check_for_duplicates(passwd_entries=passwd_entries)
+    assert len(duplicates) == 1, f"duplicate uid is not found successfully"
+
+
+def _check_for_duplicates(
```

Comment:
typically we would want to move utility functions to the plugins/utils.py file.. I'm wondering if this should not be written more generically, then it could also be used in general cases where 'list does not contain duplicates' is a requirement